packages/core/src/grpc/decorators/service.decorator.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// tslint:disable:no-any
+
+// TODO(bajtos) serviceSpec needs to be typed
+export function service(serviceSpec: any) {
+  return function(controllerCtor: Function) {
+    (controllerCtor as any).grpcService = serviceSpec;
```

Comment:
Is it a good practice to mutate the class directly instead of Reflector?

Reply:
I see two issues with Reflector:

 - It relies on having a single `reflect-metadata` package in node_modules tree, which cannot be reliably achieved.
 - It's difficult to use from JavaScript codebase.

Creating a static property on the constructor class solves both issues and it's the same approach we are already using for `Model.definition` in our repository package. I think we should eventually use this approach for storing OpenAPI spec too.

Thoughts?

cc @kjdelisle @virkt25

Reply:
+1, for the first reason alone; if others are making use of reflect-metadata, which is entirely likely, it could impact us.